### DISCUSSION TITLE
Valheim Server on FullTime

### DISCUSSION THREAD

**Discussion by WindLother** [GameServerManagers/LinuxGSM#3281] [Q&A] [answered]
Hello there, I got to setting up my Valheim server sucessfully but... from time to time the server keeps shutting down and I have to start the server again.
Is there any configuration or param to keep the server running full time? (It's a dedicated server so it's not internet connection).

**Comment by rconjoe** [comment]
Could you include some more information such as your vps configuration, hosting provider, the results of `./vhserver postdetails`, and whatever else you might think is helpful? It's abnormal behavior for the server to turn off outside of you turning it off, if the vm itself is restarting you could write* cronjobs to ensure it comes online on restart

**Reply by WindLother** [reply]
The postdetails: http://termbin.com/94zg

Hosting provider is locaweb, a famous vps provider in my country.

The vps is not restarting since I let the window opened all the time and whenever I remote access the server, using the command ./vhserver details, the server is offline (I get feedback that the players are playing and the server shuts down).

**Reply by rconjoe** [reply]
I'm leaning more towards an issue coming from the game itself, is there anything of note in your logs? 

Check `gameserver/log/console/vhserver*.log`, you should be able to find where players are connecting to the server and, hopefully not far below, that a reason why the server is disconnecting

**Reply by WindLother** [reply]
Looking through the logs I found this, but I have no clue about what could be causing the server disconnect:

> **_02/18/2021 17:50:21: World saved ( 4947.747ms )
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> 02/18/2021 18:00:15:  Connections 0 ZDOS:873536  sent:0 recv:0
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> Unloading 0 Unused Serialized files (Serialized files now loaded: 0)
> 
> Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 1116725.
> Total: 227266.694482 ms (FindLiveObjects: 162.823066 ms CreateObjectMapping: 51.886228 ms MarkObjects: 226968.664764 ms  DeleteObjects: 83.313444 ms)
> 
> 02/18/2021 18:04:03: Game server disconnected
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> 02/18/2021 18:04:21: Game server connected
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> 02/18/2021 18:14:03:  Connections 0 ZDOS:873536  sent:0 recv:0
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> Unloading 0 Unused Serialized files (Serialized files now loaded: 0)
> 
> Unloading 0 unused Assets to reduce memory usage. Loaded Objects now: 1116725.
> Total: 534648.221440 ms (FindLiveObjects: 168.749824 ms CreateObjectMapping: 49.900748 ms MarkObjects: 534351.931785 ms  DeleteObjects: 77.632794 ms)
> 
> 02/18/2021 18:22:58: Game server disconnected
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
> 
> 02/18/2021 18:22:58: Game server connected failed
>  
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)_**

**Reply by rconjoe** [reply]
These logs are typical except for the unusually long asset unload times, and the fact that they aren't actually unloading anything at all. Something is going wrong with Valheim itself, have you tried this setup from scratch on a new vm?

**Comment by wldwalker** [comment, marked as answer]
I think you're running out of memory. 

Mem:       total  used   free   cached  available
Physical:  3.9GB  3.6GB  81MB   195MB   81MB

**Reply by rconjoe** [reply]
Would make the most sense.

**Reply by WindLother** [reply]
Yeah I think that makes sense too, so 4GB RAM may not be enough to run the server

**Reply by cinnes** [reply]
This is somewhat anecdotal, I've been running vhserver on a 4GB droplet on digitalocean for a while without issues. But after updating the server a few days ago I noticed that the process is now eating more RAM and using less CPU when relatively idle. Here are my monitoring graphs:

![image](https://user-images.githubusercontent.com/19338843/108553312-65288480-72ea-11eb-976f-c54a037abc2f.png)

Previously 4GB was fine for the amount of people I had on my server (1-6, staying ~60% usage) but I imagine now I'd be getting dangerously close to hitting 100% if I had more people on.

**Reply by WindLother** [reply]
> This is somewhat anecdotal, I've been running vhserver on a 4GB droplet on digitalocean for a while without issues. But after updating the server a few days ago I noticed that the process is now eating more RAM and using less CPU when relatively idle. Here are my monitoring graphs:
> 
> ![image](https://user-images.githubusercontent.com/19338843/108553312-65288480-72ea-11eb-976f-c54a037abc2f.png)
> 
> Previously 4GB was fine for the amount of people I had on my server (1-6, staying ~60% usage) but I imagine now I'd be getting dangerously close to hitting 100% if I had more people on.

This is the same amount of players I have in my server. I found strange that 4GB wasn't being enough and an 8GB VPS is way too expensive just for a server to play with friends.

**Reply by dgibbs64** [reply]
new changelog for Valheim

0.145.6
* Disable google analytics
* Updated Unity engine
* Forest crypt fixes
* Credits updated
* Save system returned to old-new
* Pickables sync fix (stones,carrots etc)
*** Dedicated server CPU usage fix (Framerate limiter)**
* Hearth smoke building fix
* Save system tweaks
* Console "save" function can be used remotely if player is added to admin list
* Autosave every 20min (instead of 30min)
* Manual join IP function (dedicated servers only)
* Extra tombstone checks ( to prevent them from disappearing )
* Better broken world handling ( crash/exit instead of loading an empty world )
* Localization fixes

**Reply by cinnes** [reply]
@dgibbs64 👍 perhaps they're pushing more into memory for the CPU usage fix, which is having an unfortunate side effect that some servers are pushing into the next tier

**Comment by dgibbs64** [comment]
side note if you havnt make sure the monitor cron is setup

**Reply by WindLother** [reply]
Can I configure the cron to get the server up after shutting down?
Like a checkpoint every 2 minutes?

**Reply by dgibbs64** [reply]
@WindLother check out the docs https://docs.linuxgsm.com there is a section on how to get things running on boot :)
